e2e.bats

```diff
@@ -1,19 +1,38 @@
 #!/usr/bin/env bats
 
-@test "reject because name is on deny list" {
-  run kwctl run policy.wasm -r test_data/ingress.json --settings-json '{"denied_names": ["foo", "tls-example-ingress"]}'
+@test "reject because allowedTypes is empty" {
+  run kwctl run policy.wasm -r test_data/request-pod-volumes.json \
+    --settings-json \
+    '{ "allowedTypes": [] }'
 
   # this prints the output when one the checks below fails
   echo "output = ${output}"
 
   # request rejected
   [ "$status" -eq 0 ]
   [ $(expr "$output" : '.*allowed.*false') -ne 0 ]
-  [ $(expr "$output" : ".*The 'tls-example-ingress' name is on the deny list.*") -ne 0 ]
+  [ $(expr "$output" : ".*No volume type is allowed.*") -ne 0 ]
 }
 
-@test "accept because name is not on the deny list" {
-  run kwctl run policy.wasm -r test_data/ingress.json --settings-json '{"denied_names": ["foo"]}'
+@test "reject because types not present" {
```

Comment:
Maybe add another test covering a invalid settings (the usage of `*` together with something else)?

Reply:
That's a good idea. Thing is, we already have the test for validating the settings: https://github.com/kubewarden/psp-volumes-policy/blob/6ea8abb85e986527d5d9bdb5e6bbd4c8585d2614/settings_test.go#L60-L83

If I add the following test to `e2e.bats`, weirdly, kwctl allows it:
```@test "reject because mix of '*' and other types" {
  run kwctl run policy.wasm -r test_data/request-pod-volumes.json \
    --settings-json \
    '{ "allowedTypes": ["projected", "hostPath", "*"] }'

  # this prints the output when one the checks below fails
  echo "output = ${output}"

  # request rejected
  [ "$status" -eq 0 ]
  [ $(expr "$output" : '.*allowed.*false') -ne 0 ]
  [ $(expr "$output" : "") -ne 0 ]
}
```
bats output:
```
bats e2e.bats
 ✓ reject because allowedTypes is empty 
 ✓ reject because types not present 
 ✗ reject because mix of '*' and other types
   (in test file e2e.bats, line 40)
     `[ "$status" -eq 1 ]' failed
   output = {"uid":"1299d386-525b-4032-98ae-1949f69f9cfc","allowed":true}
 ✓ accept all types 
 ✓ accept pods with no volumes 
 ✓ accept pods with correct types 

6 tests, 1 failure
```

Here it is standalone:
```
$ kwctl run policy.wasm -r test_data/request-pod-volumes.json --settings-json '{ "allowedTypes": ["*", "projected", "hostPath"] }'
{"uid":"1299d386-525b-4032-98ae-1949f69f9cfc","allowed":true}
```

This looks like an error on kwctl to me?

Reply:
Indeed, it was a problem that I was expecting a different json in NewSettingsFromValidateSettingsPayload(). Thanks!

Fixed it in https://github.com/kubewarden/psp-volumes-policy/pull/1/commits/6956a0b974675256f007ac433e83576a1f8269f0 (with its on test, now using correct assumption), and added an e2e test for it in https://github.com/kubewarden/psp-volumes-policy/pull/1/commits/dd099dd6703f30d2bac461baf6ea695633d755a1.